draft-ietf-tls-sni-encryption.md

```diff
@@ -319,6 +319,49 @@ QUESTION: Do we need a revocation mechanism? What if a fronting service obtains
 delegation token, and then becomes untrustable for some other reason? Or is it
 sufficient to just use short TTL?
 
+# Supporting multiple protocols
+
+The SNI encryption requirement do not stop with HTTP over TLS. Multiple other
+applications currently use TLS, including for example SMTP [@?RFC5246], DNS [@?RFC7858], 
+or XMPP [@?RFC7590]. These applications too will benefit of SNI encryption.
+HTTP only methods like those described in (#httpfrontingtunnels)
+would not apply there. In fact, even for the HTTPS case, the HTTPS tunneling service
+described in (#httpfrontingtunnels) is compatible with HTTP 1.0 and HTTP 1.1, 
+but interacts awkwardly with the multiple streams feature of HTTP 2.0 [@?RFC7540].
+This points to the need of an application agnostic solution, that would be
+implemented fully in the TLS layer.
+
+## Hiding the Application Layer Protocol Negotiation
+
+The Application Layer Protocol Negotiation (ALPN) parameters of TLS
+allow implementations to negotiate the application layer protocol used on
+a given connection. TLS provides the ALPN values in clear text during the
+initial handshake. Exposing the ALPN does not create the same
```

Comment:
I think you want to say "While exposing"

Reply:
OK.